crystal_diffusion/samplers/predictor_corrector_position_sampler.py

```diff
@@ -24,26 +24,30 @@ def __init__(self, number_of_discretization_steps: int, number_of_corrector_step
         self.number_of_discretization_steps = number_of_discretization_steps
         self.number_of_corrector_steps = number_of_corrector_steps
 
-    def sample(self, number_of_samples: int, device=torch.device) -> torch.Tensor:
+    def sample(self, number_of_samples: int, device: torch.device, unit_cell: torch.Tensor) -> torch.Tensor:
         """Sample.
 
         This method draws a sample using the PR sampler algorithm.
 
         Args:
             number_of_samples : number of samples to draw.
             device: device to use (cpu, cuda, etc.). Should match the PL model location.
+            unit_cell: unit cell definition in Angstrom.
+                Tensor of dimensions [batch_size, spatial_dimension, spatial_dimension]
 
         Returns:
             position samples: position samples.
         """
         x_ip1 = map_positions_to_unit_cell(self.initialize(number_of_samples)).to(device)
-        logger.info("Starting position sampling")
+
+        if unit_cell.dim() == 2:  # no batchsize for unit cell, just repeat
```

Comment:
This contradicts the docstring, which indicates that the `unit_cell` should already have a "batch_size" dimension. 
Here, the "batch_size" is actually the variable `number_of_samples`.  I suggest that either:
1. you modify the docstring to indicate that `unit_cell` is a `[spatial_dimension, spatial_dimension]` tensor, making this officially a 'fixed box' sampler. You can then broadcast the unit cell tensor to dimension `[number_of_samples, spatial_dimension, spatial_dimension]`: no need to infer the batch dimension from `x_ip1`.

or 

2. you broadcast the `unit_cell` tensor to the right dimension outside of this method and here just assert/check that the `unit_cell` tensor has the right dimensions. The sampler would then be ready for the glorious future where we start sampling box sizes.